compute/dstess.c

```diff
@@ -0,0 +1,1333 @@
+   }
+
+   if (AV != test_vector) free(AV);
+   if (zeros == n) return(DBL_MAX);
+   return(norm);
+} // end eigp_error.
+
+
+//-----------------------------------------------------------------------------
+// Thread work.
+// We are given the WorkStack; we only exit when eigenvalues == finished.  The
+// ToDo chain works like a stack, we pop Brackets to work on.  Each Bracket is
+// bisected down to one eigenvalue; right-halves are pushed back on the ToDo
+// stack as future work to do.
+//
+// For range=PlasmaRangeI: We still bisect on range values, but the conditions
+// change. We begin with the full range Gerschgorin [low, hi+ulp). We compute
+// nLT_low, nLT_hi, but we need the indices il, iu. So these nLT_x values are
+// the conditions: The smallest (1 relative) index the bracket can contain is
+// nLT_low+1. e.g. if nLT_low = 0, then index 1 may be in the bracket.  if
+// nLT_low=7, then index 8 may be in the bracket.
+//
+// The largest (1 relative) index the bracket can contain is nLT_hi: If
+// nLT_hi=n, then the nth eigenvalue might be in the bracket. So the bracket
+// contains indices [nLT_low+1, nLT_hi]. How does that intersect [il, iu]?
+//
+// Low end: If iu < nLT_low+1, then [il,iu] is out of Bracket.  Hi end: If il >
+// nLT_hi, then [il,iu] is out of Bracket.  Otherwise there is overlap. We find
+// the midpoint, and compute nLT_midp.
+//
+// If nLT_midp were the new high (for the left bracket), then if il > nLT_midp,
+// we can discard the left bracket. 
+//
+// If nLT_midp were the new low (for the right bracket), then if iu <
+// nLT_midp+1, we can discard the right bracket. 
+//
+// The number of eigenvalues in the bracket is still (nLT_hi - nLT_low). if
+// that is ever zero, we can discard the bracket.
+//
+// Stage: Init: establish nLT_low, nLT_hi, numEV in the bracket.
+//
+// Stage: Bisection. We use Bisection to divide the range until lowerBound and
+// upperBound cannot be averaged. (The result is one or the other).  If we
+// subdivide and the two halves both have eigenvalues, we create a new bracket
+// and (Critical) add it to the ToDo stack, with Bisection as the workstage.
+//
+// Once a range can't be subdivided (the UpperBound=LowerBound+ULP); we store
+// the lowerBound as the eigenvalue and numEV as the multiplicity in the
+// caller's array (pVal, pMul), and enter the next Stage, GetVector.
+//
+// Stage: GetVector. At this point we know the index of the eigenpair within
+// the user's arrays. If PlasmaVec, we find the eigenvector using the LAPACK
+// routine DSTEIN, storing it in its proper place in pVec.  Then we add numEV
+// to the count of finished eigenpairs, and free the bracket.
+//
+// The most comparable serial LAPACK routine is DSTEVX.
+//
+// Once all thread work is complete, the code will condense these arrays to
+// just the actual number of unique eigenvalues found.
+//-----------------------------------------------------------------------------
+
+void thread_work(WorkStack* Stack) {
```

Comment:
This is an external name and it's too generic. And it's large enough for it's own name and source code file.